packages/ipfs/test/sharness/t0054-dag-car-import-export.sh

```diff
@@ -0,0 +1,204 @@
+
+  cat > basic_import_expected <<EOE
+Pinned root${tab}bafkqaaa${tab}success
+Pinned root${tab}bafy2bzaceaxm23epjsmh75yvzcecsrbavlmkcxnva66bkdebdcnyw3bjrc74u${tab}success
+Pinned root${tab}bafy2bzaced4ueelaegfs5fqu4tzsh6ywbbpfk3cxppupmxfdhbpbhzawfw5oy${tab}success
+EOE
+
+  cat >naked_root_import_json_expected <<EOE
+{"Root":{"Cid":{"/":"bafy2bzaceaxm23epjsmh75yvzcecsrbavlmkcxnva66bkdebdcnyw3bjrc74u"},"PinErrorMsg":""}}
+{"Root":{"Cid":{"/":"bafy2bzaced4ueelaegfs5fqu4tzsh6ywbbpfk3cxppupmxfdhbpbhzawfw5oy"},"PinErrorMsg":""}}
+EOE
+
+
+  test_expect_success "basic import" '
+    do_import 0 \
+      ../t0054-dag-car-import-export-data/combined_naked_roots_genesis_and_128.car \
+      ../t0054-dag-car-import-export-data/lotus_testnet_export_128_shuffled_nulroot.car \
+      ../t0054-dag-car-import-export-data/lotus_devnet_genesis_shuffled_nulroot.car \
+    > basic_import_actual
+  '
+
+  test_expect_success "basic import output as expected" '
+    test_cmp_sorted basic_import_expected basic_import_actual
+  '
+
+  test_expect_success "basic fetch+export 1" '
+    ipfsi 1 dag export bafy2bzaced4ueelaegfs5fqu4tzsh6ywbbpfk3cxppupmxfdhbpbhzawfw5oy > reexported_testnet_128.car
+  '
+  test_expect_success "export of shuffled testnet export identical to canonical original" '
+    test_cmp reexported_testnet_128.car ../t0054-dag-car-import-export-data/lotus_testnet_export_128.car
+  '
+
+  test_expect_success "basic fetch+export 2" '
+    ipfsi 1 dag export bafy2bzaceaxm23epjsmh75yvzcecsrbavlmkcxnva66bkdebdcnyw3bjrc74u > reexported_devnet_genesis.car
+  '
+  test_expect_success "export of shuffled devnet export identical to canonical original" '
+    test_cmp reexported_devnet_genesis.car ../t0054-dag-car-import-export-data/lotus_devnet_genesis.car
+  '
+
+  test_expect_success "pinlist on node1 still empty" '
+    [[ -z "$( ipfsi 1 pin ls )" ]]
+  '
+
+  test_expect_success "import/pin naked roots only, relying on local blockstore having all the data" '
+    ipfsi 1 dag import --enc=json ../t0054-dag-car-import-export-data/combined_naked_roots_genesis_and_128.car \
+      > naked_import_result_json_actual
+  '
+
+  test_expect_success "naked import output as expected" '
+    test_cmp_sorted naked_root_import_json_expected naked_import_result_json_actual
+  '
+
+  reset_blockstore 0
+  reset_blockstore 1
+
+  mkfifo pipe_testnet
+  mkfifo pipe_devnet
+
+  test_expect_success "fifo import" '
+    (
+        cat ../t0054-dag-car-import-export-data/lotus_testnet_export_128_shuffled_nulroot.car > pipe_testnet &
+        cat ../t0054-dag-car-import-export-data/lotus_devnet_genesis_shuffled_nulroot.car > pipe_devnet &
+
+        do_import 0 \
+          pipe_testnet \
+          pipe_devnet \
+          ../t0054-dag-car-import-export-data/combined_naked_roots_genesis_and_128.car \
+        > basic_fifo_import_actual
+        result=$?
+
+        wait || true	# work around possible trigger of a bash bug on overloaded circleci
+        exit "$result"
+    )
+  '
```

Comment:
@rvagg the exercise of the GC-lock and the import of FIFOs may not be something you are too inteerested in testing within js-ipfs. Just raising it here as it was a very important part of making 🗡️ viable.